mdsuite/graph_modules/molecular_graph.py

```diff
@@ -60,143 +70,141 @@ def __init__(
 
-    @staticmethod
-    def get_neighbour_list(positions: tf.Tensor, cell: list = None) -> tf.Tensor:
-        """
-        Generate the neighbour list
+                .. code-block::
 
-        Parameters
-        ----------
-        positions: tf.Tensor
-            Tensor with shape (number_of_configurations, n_atoms, 3)
-            representing the coordinates
-        cell: list
-            If periodic boundary conditions are used, please supply the cell
-            dimensions, e.g. [13.97, 13.97, 13.97]. If the cell is provided
-            minimum image convention will be applied!
-        batch_size: int
-            Has to be evenly divisible by the the number of configurations.
+                   {'emim': {'smiles': 'CCN1C=C[N+](+C1)C', 'amount': 20, "cutoff": 1.7},
+                   'PF6': {'smiles': 'F[P-](F)(F)(F)(F)F', 'amount': 20, "cutoff": 1.7}}
 
-        Returns
-        -------
-        generator object which results all distances for the current batch of
-        time steps
+                or:
 
-        To get the real r_ij matrix for one time_step you can use the following:
-            r_ij_mat = np.zeros((n_atoms, n_atoms, 3))
-            r_ij_mat[np.triu_indices(n_atoms, k = 1)] = get_neighbour_list(``*args``)
-            r_ij_mat -= r_ij_mat.transpose(1, 0, 2)
+                .. code-block::
 
-        """
-        r_ij_matrix = tf.reshape(positions, (1, len(positions), 3)) - tf.reshape(
-            positions, (len(positions), 1, 3)
-        )
-        if cell:
-            r_ij_matrix -= tf.math.rint(r_ij_matrix / cell) * cell
-        return tf.norm(r_ij_matrix, ord="euclidean", axis=2)
+                   {'emim': {'reference': {'C': 6, 'N': 2, 'H': 12}}, 'amount': 20},
+                   'PF6': {'reference': {'P': 1, 'F': 6}, 'amount': 20, "cutoff": 1.7}}
 
-    def build_smiles_graph(self):
-        """
-        Build molecular graphs from SMILES strings.
         """
-        mol = read_smiles(self.smiles_string, explicit_hydrogen=True)
-        data = mol.nodes
-        species = {}
-        for i in range(len(data)):
-            item = data[i].get("element")
-            if item in species:
-                species[item] += 1
-            else:
-                species[item] = 0
-                species[item] += 1
+        self.experiment = experiment
+        self.molecule_name = molecule_input_data.name
+        self.database = Database(self.experiment.database_path / "database.hdf5")
+        self.cutoff = molecule_input_data.cutoff
+        self.n_molecules = molecule_input_data.amount
+        self.mol_pbc = molecule_input_data.mol_pbc
 
-        return mol, species
+        if self.mol_pbc:
+            self.reference_property = "Positions"
+        else:
+            self.reference_property = "Unwrapped_Positions"
 
-    @staticmethod
-    def _apply_system_cutoff(tensor: tf.Tensor, cutoff: float) -> tf.Tensor:
+        if isinstance(molecule_input_data.reference_configuration, int):
```

Comment:
is reference_configuration a reference_configuration_idx?

Reply:
it is, I though from a user interface POV that dropping idx would be simpler, but if you disagree it can be changed.

Reply:
in my head, the idx immediately tells me I have to provide an ``int``. My first thought when reading ``reference_configuration`` is a (n_partcl, n_dims) array

Reply:
No problem I will update it.

Reply:
resolved in 7cac760